src/Illuminate/Mail/Message.php

```diff
@@ -344,12 +346,16 @@ public function embed($file)
                 function ($path) use ($file) {
                     $cid = $file->as ?? Str::random();
 
-                    $this->message->embedFromPath($path, $cid, $file->mime);
+                    $this->message->addPart(
+                        (new DataPart(new File($path), $cid, $file->mime))->asInline()
+                    );
 
                     return "cid:{$cid}";
                 },
                 function ($data) use ($file) {
-                    $this->message->embed($data(), $file->as, $file->mime);
+                    $this->message->addPart(
+                        (new DataPart($data(), $file->as, $file->mime))->asInline()
```

Comment:
This is the only part where I am unsure:
I think `$data()` will return a `string` since it is the `dataStrategy` instead of the `fileStrategy`.

But I could not find any usage of this in the framework.

Reply:
If you aren't sure, we will need to wait for @timacdonald to review this.

Reply:
@Jubeki you are correct that `$data()` returns a string.